unit/solvers/smt2_incremental/construct_value_expr_from_smt.cpp

```diff
@@ -7,18 +7,70 @@
 #include <solvers/smt2_incremental/smt_terms.h>
 #include <solvers/smt2_incremental/smt_to_smt2_string.h>
 
+#include <util/arith_tools.h>
+#include <util/bitvector_types.h>
+#include <util/mp_arith.h>
 #include <util/std_expr.h>
 #include <util/std_types.h>
 
+static mp_integer power2(unsigned exponent)
+{
+  mp_integer result;
+  result.setPower2(exponent);
+  return result;
+}
+
+/// Returns the maximum integer value which can be stored in \p bits as an
+/// unsigned integer.
+static mp_integer max_int(const std::size_t bits)
+{
+  return power2(bits) - 1;
+}
+
 TEST_CASE("Value expr construction from smt.", "[core][smt2_incremental]")
 {
   optionalt<smt_termt> input_term;
   optionalt<exprt> expected_result;
 
   using rowt = std::pair<smt_termt, exprt>;
+
+  // clang-format off
+#define UNSIGNED_BIT_VECTOR_TESTS(bits)                                        \
```

Comment:
Ohh, I like that. 👍🏻 

Better for self-documenting code than the comment based approach we had.